client/web/compose/src/components/Admin/Module/UniqueValues.vue

```diff
@@ -0,0 +1,179 @@
+                  {{ $t('valueModifiers') }}
+                </th>
+                <th style="width: 250px;">
+                  {{ $t('multiValues') }}
+                </th>
+              </tr>
+            </thead>
+            <tbody v-if="rule.constraints">
+              <tr
+                v-for="(constraint, consIndex) in rule.constraints"
+                :key="`constraint-${consIndex}`"
+              >
+                <td>{{ constraint.attribute }}</td>
+                <td>{{ constraint.kind }}</td>
+                <td>
+                  <b-form-select
+                    v-model="constraint.modifier"
+                    :options="modifierOptions"
+                    size="sm"
+                  />
+                </td>
+                <td>
+                  <b-form-select
+                    v-model="constraint.multiValue"
+                    :options="multiValueOptions"
+                    :disabled="!constraint.isMulti"
+                    size="sm"
+                  />
+                </td>
+                <td class="text-center">
+                  <c-input-confirm
+                    variant="link"
+                    size="sm"
+                    button-class="text-dark px-0 text-right"
+                    @confirmed="rule.constraints.splice(consIndex, 1)"
+                  />
+                </td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <div class="mt-3">
+          <b-row>
+            <b-col
+              sm="12"
+              md="6"
+            >
+              <b-form-group>
+                <b-input-group class="w-100">
+                  <vue-select
+                    v-model="rule.currentField"
+                    :placeholder="$t('searchFields')"
+                    :options="filterFieldOptions(rule)"
+                    style="width: 80%"
+                    class="bg-white"
+                  />
+
+                  <b-input-group-append style="width: 20%">
+                    <b-button
+                      size="sm"
+                      variant="light"
+                      class="px-4"
+                      @click="() => updateRuleConstraint(rule)"
+                    >
+                      {{ $t('add') }}
+                    </b-button>
+                  </b-input-group-append>
+                </b-input-group>
+              </b-form-group>
+            </b-col>
+          </b-row>
+        </div>
+      </div>
+      <hr>
+    </div>
+  </div>
+</template>
+
+<script>
+import { compose } from '@cortezaproject/corteza-js'
+import { VueSelect } from 'vue-select'
+
+export default {
+  i18nOptions: {
+    namespaces: 'module',
+    keyPrefix: 'edit.config.uniqueValues',
+  },
+
+  components: {
+    VueSelect,
+  },
+
+  props: {
+    module: {
+      type: compose.Module,
+      required: true,
+    },
+  },
+
+  computed: {
+    rules () {
+      return this.module.config.recordDeDup.rules
```

Comment:
This should be a computed similar to page block options, get set and emit events. since prop updating directly isn't really a good thing